payu/git_utils.py

```diff
@@ -124,7 +124,16 @@ def remote_branches_dict(self) -> Dict[str, git.Head]:
         objects"""
         branch_names_dict = {}
         for remote in self.repo.remotes:
-            remote.fetch()
+            try:
+                remote.fetch()
+            except git.exc.GitCommandError:
+                warnings.warn(
+                    f"Failed to fetch from remote repository: {remote.name} " +
```

Comment:
I know it might make it a bit verbose, but is it worth explaining why `payu` was trying to do this so the user knows they an either safely ignore it, or if fix it if they want that functionality?